### PR TITLE
Refactor methods for generating/parsing/validating IDs to a helper class

### PR DESCRIPTION
**Pull request by HaloFour**
Pondering moving the logic for generating, validating and parsing of trace and span IDs into separate helper class so that `SpanId` can be less cluttered.  I don't know that I'm entirely happy with this but I thought I'd get some feedback.

### CODE REVIEW

File: money-api/src/main/java/com/comcast/money/api/IdGenerator.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+    /**
+     * Generates a random trace ID.
+     */
+    public static String generateRandomTraceId() {
+        long hi, lo;
+        Random random = threadLocalRandom.get();
+        do {
+            hi = random.nextLong();
+            lo = random.nextLong();
```
Comment (ptravers) [inline money-api/src/main/java/com/comcast/money/api/IdGenerator.java:47]: do we know how random this is?

**Review by ptravers — COMMENTED** [review]
looks good. just one q around the safety of nextLong

File: money-api/src/main/java/com/comcast/money/api/IdGenerator.java (hunk 1 above)
Comment (HaloFour) [reply money-api/src/main/java/com/comcast/money/api/IdGenerator.java:47]: Does it help that this is the implementation that the OTel SDK uses?

https://github.com/open-telemetry/opentelemetry-java/blob/master/sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java#L31

I did notice that `ThreadLocalRandom` is actually a class provided by Java 8, I can at least switch the impl to using that.

Comment (ptravers) [reply money-api/src/main/java/com/comcast/money/api/IdGenerator.java:45]: Yeah I would be okay with defaulting to what otel does here. I won't have time to dig in to this tonight. But I have set aside some time early tomorrow morning to double check

**Review by ptravers — COMMENTED** [review]
Will double check the randomness safety

**Comment by ptravers** [comment]
I like the refactor as it exposes some critical logic

**Comment by ptravers** [comment]
this is likely preferable to minimize contention and garbage collection

http://thoughtfuljava.blogspot.com/2012/09/prefer-threadlocalrandom-over-random.html

**Review by ptravers — APPROVED** [review]

**Pull request merged**